test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
-      user = users(:joan_doe)
-      login_as user
-      project2 = projects(:project2)
-      visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
-      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 3, count: 3,
-                                                                           locale: @user.locale))
-      within '#samples-table table tbody' do
-        assert_selector 'tr', count: 3
-        all('input[type=checkbox]').each { |checkbox| checkbox.click unless checkbox.checked? }
-      end
-      click_link I18n.t('projects.samples.index.transfer_button'), match: :first
-      within('span[data-controller-connected="true"] dialog') do
-        within %(turbo-frame[id="list_selections"]) do
-          samples = @project.samples.pluck(:puid, :name)
-          samples.each do |sample|
-            assert_text sample[0]
-            assert_text sample[1]
-          end
-        end
-        find('input#select2-input').click
-        find("button[data-viral--select2-primary-param='#{project2.full_path}']").click
-        click_on I18n.t('projects.samples.transfers.dialog.submit_button')
-      end
-    end
+      ### VERIFY START ###
+      # flash msg
+      assert_text I18n.t('projects.samples.transfers.create.success')
+      # originating project no longer has samples
+      assert_text I18n.t('projects.samples.index.no_samples')
 
-    test 'sample transfer project listing should be empty for maintainer if no other projects in hierarchy' do
-      user = users(:user28)
-      login_as user
-      namespace = groups(:group_hotel)
-      project2 = projects(:projectHotel)
-      Project.reset_counters(project2.id, :samples_count)
-      visit namespace_project_samples_url(namespace, project2)
-      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 1, count: 1,
-                                                                           locale: @user.locale))
+      # destination project received transferred samples
+      visit namespace_project_samples_url(@namespace, @project2)
       within '#samples-table table tbody' do
-        assert_selector 'tr', count: 1
-        all('input[type=checkbox]').each { |checkbox| checkbox.click unless checkbox.checked? }
-      end
-      click_link I18n.t('projects.samples.index.transfer_button'), match: :first
-      within('span[data-controller-connected="true"] dialog') do
-        within %(turbo-frame[id="list_selections"]) do
-          samples = project2.samples.pluck(:puid, :name)
-          samples.each do |sample|
-            assert_text sample[0]
-            assert_text sample[1]
-          end
+        samples.each do |sample|
+          assert_text sample[0]
+          assert_text sample[1]
         end
-        assert_no_selector 'option'
       end
+      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 20, count: 23,
```

Comment:
**nitpick (blocking):** Can you assert for this above the samples assertions, that should reduce flakes

Reply:
Fixed in 9163062d5b74448effd6297caf34663c72cb36de